Source/diablo.cpp

```diff
@@ -150,6 +150,60 @@ extern void plrctrls_after_check_curs_move();
 extern void plrctrls_every_frame();
 extern void plrctrls_after_game_logic();
 
+bool TryOpenDungeon(bool useCursorPosition)
+{
+	if (leveltype != DTYPE_TOWN)
+		return false;
+
+	Item &holdItem = MyPlayer->HoldItem;
+	WorldTilePosition &position = MyPlayer->position.tile;
+
+	// Common function to attempt opening based on input method.
+	auto attemptOpen = [&]() -> bool {

```

Comment:
What need does the arrow function serve, it seems it should work the same with the code directly instead.